src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldType.java

```diff
@@ -51,7 +52,12 @@ public KNNVectorFieldType(String name, Map<String, String> metadata, VectorDataT
 
     @Override
     public ValueFetcher valueFetcher(QueryShardContext context, SearchLookup searchLookup, String format) {
-        throw new UnsupportedOperationException("KNN Vector do not support fields search");
+        return new ArraySourceValueFetcher(name(), context) {
```

Comment:
if this is static can we create instance once and reuse it instead of creating new ArraySourceValueFetcher on every call?